src/main/kotlin/org/rust/ide/refactoring/extractEnumVariant/impl.kt

```diff
@@ -0,0 +1,357 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.extractEnumVariant
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.ide.refactoring.RsInPlaceVariableIntroducer
+import org.rust.lang.core.ARBITRARY_ENUM_DISCRIMINANT
+import org.rust.lang.core.FeatureAvailability
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.types.consts.Const
+import org.rust.lang.core.types.consts.CtConstParameter
+import org.rust.lang.core.types.infer.TypeVisitor
+import org.rust.lang.core.types.infer.hasCtConstParameters
+import org.rust.lang.core.types.infer.hasTyTypeParameters
+import org.rust.lang.core.types.ty.Ty
+import org.rust.lang.core.types.ty.TyTypeParameter
+import org.rust.lang.core.types.type
+
+
+sealed class VariantElement(val toBeReplaced: PsiElement, val factory: RsPsiFactory) {
```

Comment:
Looks like almost everything in this file (except `findApplicableContext`) is implementation details of `RsExtractEnumVariantProcessor` and I suggest keeping them into single file and make everything private (if possible). We don't use implementation details anywhere (including tests) so no need to provide some items to global scope